test/sys/test_termios.rs

```diff
@@ -19,3 +42,93 @@ fn test_tcgetattr() {
         }
     }
 }
+
+// Test modifying output flags
+#[test]
+fn test_output_flags() {
+    // Open one pty to get attributes for the second one
+    let mut termios = {
+        let pty = openpty(None, None).unwrap();
+        assert!(pty.master > 0);
+        assert!(pty.slave > 0);
+        let termios = tcgetattr(pty.master).unwrap();
+        close(pty.master).unwrap();
```

Comment:
This is a double close.  See issue #659.  There are a few more instances of it below.